src/PMGDQuery.cc

```diff
@@ -347,10 +347,8 @@ void PMGDQuery::parse_query_constraints(const Json::Value& constraints,
         const Json::Value& predicate = *it;
         const std::string& key = it.key().asString();
 
-        // Will either have 2 or 4 arguments
-        assert(predicate.isArray());
-        assert(predicate.size() == 2 || predicate.size() == 4);
-
+        // Will either have 2 or 4 arguments as verified when parsing
+        // JSON
```

Comment:
That check was already happening at the JSON level. I added the array size check right after that. So I figured there was no need for these asserts anymore. Plus, I don't think the server should crash because JSON is malformed for a query. So the asserts were wrong in general